### PR TITLE
WIP: JuliaCon 2017 blog post

### PR DESCRIPTION
**Pull request by ranjanan**
Needs a ton of editing, and a little more material.

### CODE REVIEW

**Comment by ararslan** [comment]
Wow, thanks for doing this! :D

File: blog/_posts/2017-06-30-juliacon.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+It’s a year later and I’m back: this time to the West Coast for the next edition of JuliaCon.
```
Comment (ararslan) [inline blog/_posts/2017-06-30-juliacon.md]: Needs a header:
```
---
layout: post
title:  West Coast is Best Coast for JuliaCon
author: Ranjan Arantharaman
---
```
(with the title replaced with whatever you'd like to call it)

File: blog/_posts/2017-06-30-juliacon.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+the computation. He spoke about how he had to make several improvements to the Julia compiler so as to make 
+memory access patterns more uniform and hence more efficient, and how they even used Julia packages almost as 
+is for the project. Not only was it remarkable that they crossed 1 Petaflop (which is probably the first for a 
+high productivity language), they managed to do so without writing hand tuned Ninja code. This, for me, was the 
+biggest achievement. At this point I had to run off to prepare for my second talk at the conference: Circuitscape, 
+a landscape modelling tool that was used to calculate least resistance paths across large swathes of land. I went 
+into detail about my vision for the package and how I will take it there, and hopefully earned a couple of collaborators. 
+       
+At the end of the day, there was a poster session with GSoC students presenting their work over posters. They were 
+all great. It was in particular nice to see an old intern of ours, Divyansh, do some great work on parallel graph
```
Comment (ChrisRackauckas) [inline blog/_posts/2017-06-30-juliacon.md]: Should probably elaborate more here.

**Comment by ararslan** [comment]
It would be great to provide inline links to the packages and people that are mentioned.

File: blog/_posts/2017-06-30-juliacon.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+It’s a year later and I’m back: this time to the West Coast for the next edition of JuliaCon. 
+ 
+The first day was workshop day. The first workshop was on the DifferentialEquations ecosystem, 
+championed by Chris Rackaukas. He spoke about his vision to make the ecosystem the scientist’s
```
Comment (ararslan) [inline blog/_posts/2017-06-30-juliacon.md]: Rackauckas (missing a second c)

File: blog/_posts/2017-06-30-juliacon.md (hunk 2 above)
Comment (ranjanan) [inline blog/_posts/2017-06-30-juliacon.md]: Yes, there are some places where I need to add a little content. Thanks!

**Comment by ViralBShah** [comment]
@andreasnoack Any idea on the video upload schedule? Would be nice to link them in here.

**Comment by andreasnoack** [comment]
Nah. Not sure what is really going on. I gave Giro the footage Monday and he told that they'd be working on it Tuesday and start uploading right away. Sent a reminder Thursday morning and was told that videos would be uploaded Friday but I still don't see any videos. I'll ask again.

**Comment by andreasnoack** [comment]
Some of them are up now. The template is wrong and says "JuliaCon 2015" but Giro has just replied that his assistant "will fix it" but given that is July 4 tomorrow, I'm not sure that it will happen before Wednesday.

File: blog/_posts/2017-06-30-juliacon.md (hunk 1 above)
Comment (ranjanan) [inline blog/_posts/2017-06-30-juliacon.md]: Done

**Comment by ViralBShah** [comment]
Any update on the videos?

**Comment by ranjanan** [comment]
Not all of them are up yet.

**Comment by andreasnoack** [comment]
There are still quite a few missing. I asked for an update earlier today and was told that "We're exporting a huge batch out now and should start uploading them tomorrow." However, there have been many problems with the already uploaded versions. I've e.g. disabled Jeff's talk and the Celeste talk, so we might also need to disable some of the new videos.

**Comment by ViralBShah** [comment]
Do we have a list of missing videos? And ones that need fixing?

**Comment by andreasnoack** [comment]
See https://docs.google.com/spreadsheets/d/1yQ3wAEIjGXuH0yRbA1xOzCIbLBL7nbvUTfzKRNBphFE/edit#gid=0. Half of the workshops are missing and a few of the talks. I've asked Valentin to confirm that the missing talks actually happened.

**Comment by andreasnoack** [comment]
A new version of the Celeste video is now on YouTube https://www.youtube.com/watch?v=uecdcADM3hY. It looks like the slides are now okay. The name below the video of the speaker is changed a bit late when Keno takes over but I think the talk is in a good enough shape that we can link to it now.

**Comment by ranjanan** [comment]
I've added the Celeste keynote and I've done some editing. I am not opposed to merging this and then adding any other links later (such as Jeff's overview of the type system overhaul).

File: blog/_posts/2017-06-30-juliacon.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+As usual, the conference began with a day dedicated to workshops. 
+The first [workshop](https://www.youtube.com/watch?v=7NDkpWoNiQ4) was on the 
+[DifferentialEquations](https://github.com/JuliaDiffEq/) ecosystem, 
+championed by Chris Rackauckas, who spoke about his vision to make the ecosystem the 
+scientist’s one stop shop for simulations. It certainly looked to be shaping up into a robust 
+ecosystem. Incidentally, Chris also has an army of GSoC students working on improvements to many 
+different little pieces. The next workshop I attended was the one on  
+[`Optim.jl`](https://github.com/JuliaNLSolvers/Optim.jl). Statistical learning
+ problems are often minimization problems where the objective function to minimize is your
+error function. The Optim ecosystem provides a rich variety of techniques to solve thisß
```
Comment (tkelman) [inline blog/_posts/2017-06-30-juliacon.md]: extra ß at the end here?

File: blog/_posts/2017-06-30-juliacon.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+The last day of the conference started Kathy Yelick from UC Berkeley 
+[talking about](https://www.youtube.com/watch?v=rj-WhTL_VXE) how the 
+High Performance Computing (HPC) and Big Data worlds are sort of merging, and how researchers need to find ways to 
+tackle larger amounts of scientific data. Then it was time for a lighter session with Jiahao 
+[talking about](https://www.youtube.com/watch?v=C2RO34b_oPM) 
+“how to take vector transposes seriously”. He went into 
+the history of the famous issue [#4774](https://github.com/JuliaLang/julia/issues/4774), much of 
+which involved heated discussion on the different notions of the word “vector”. The afternoon 
+started out with Stefan speaking briefly about the 
+[roadmap](https://github.com/JuliaLang/METADATA.jl/pull/10554) for Julia 1.0, assuring the
```
Comment (tkelman) [inline blog/_posts/2017-06-30-juliacon.md]: this link is about circuitscape, not the roadmap talk

**Comment by ViralBShah** [comment]
Let's merge once the couple of things @tkelman identified are fixed. Also, maybe worth having a link to the JuliaCon2017 youtube channel or something like that.

File: blog/_posts/2017-06-30-juliacon.md (hunk 4 above)
Comment (ranjanan) [inline blog/_posts/2017-06-30-juliacon.md]: Thanks, I just removed that.

File: blog/_posts/2017-06-30-juliacon.md (hunk 5 above)
Comment (ranjanan) [inline blog/_posts/2017-06-30-juliacon.md]: Thanks

**Comment by ranjanan** [comment]
@ViralBShah Good idea, i've added a link to the JuliaCon 2017 playlist.

**Comment by ViralBShah** [comment]
Would be nice to link the videos to the person's name rather than to words like "talk", "about", "act", etc. That way the package name can be linked to github, and the person's name to the talk.

**Pull request merged**